src/app/recipes/recipe.service.ts

```diff
@@ -24,11 +26,31 @@ export class RecipeService {
   ];
 
   getRecipeById(id: number) {
-    return this.recipes[id];
+    return this.recipes.filter((recipe: Recipe) => { return recipe.id === id; })[0];
```

Comment:
What does [0] mean here ? Do you have duplicated ids ? Are sure 'filter' is correct method here?

Reply:
Preparing to work with server. Id elements will not correspond to the position in the array.